src/generator/AutoRest.AzureResourceSchema/JsonSchema.cs

```diff
@@ -48,38 +49,29 @@ public class JsonSchema
         public string JsonType { get; set; }
 
         /// <summary>
-        /// Get the resource type of this JsonSchema. If this JsonSchema defines an Azure Resource,
-        /// then this value will be found in definition.properties.type.enum[0]. If this JsonSchema
-        /// does not define an Azure Resource, then this will return null.
+        /// Gets or sets the resource type of this JsonSchema.
         /// </summary>
         public string ResourceType
         {
             get
             {
-                string result = null;
-
-                if (Properties != null &&
-                    Properties.ContainsKey("type") &&
-                    Properties["type"].Enum != null)
-                {
-                    result = Properties["type"].Enum.SingleOrDefault();
-                }
-
-                return result;
+                return resourceType;
             }
             set
             {
-                if (properties == null)
+                resourceType = value;
+                if (Properties != null && Properties.ContainsKey("type"))
```

Comment:
Previously properties could be initialized in this (ResourceType) property's setter. Now it can't. Is that a regression?